### ISSUE TITLE
How did you get accurate noiseless heatmaps using D-RISE for yolo object detection?

### ISSUE THREAD

**Issue by arynbarman** [RuoyuChen10/objectdetection-saliency-maps#6]
Hi, I ran your code for D-RISE with Yolo and was getting noisy and inaccurate heatmaps in contrast to yours, which were accurate to objects detected in bounding boxes. Could you share which checkpoint and yolo configuration you used or any other method which you have used to generate the heatmaps.

**Comment by RuoyuChen10** [comment]
> Hi, I ran your code for D-RISE with Yolo and was getting noisy and inaccurate heatmaps in contrast to yours, which were accurate to objects detected in bounding boxes. Could you share which checkpoint and yolo configuration you used or any other method which you have used to generate the heatmaps.

It may take a few days because I have been busy recently. In addition, some suggestions are given to you. The location may be inaccurate due to too many patches of mask segmentation. Try to control it at 16*16.

**Comment by arynbarman** [comment]
Hi, as suggested by you, I varied the patch sizes and number of patches but that was not the reason for poor heatmaps.

In drise-yolov3.py when the function call to generate_saliency_map is made the target class index was hardcoded to 1.

    feat = model.extract_feat(data['img'][0].cuda())
    res = model.bbox_head.simple_test(
        feat, data['img_metas'][0], rescale=True)
    target_box = res[0][0][args.bbox_index][:-1].cpu().detach().numpy().astype(np.int32)


    saliency_map = generate_saliency_map(model,
                                        image,
                                        target_class_index=1, #hardcoded
                                        target_box=target_box,
                                        prob_thresh=0.5,
                                        grid_size=(16, 16),
                                        n_masks=1000)


So, on passing the target class index with the index of the predicted class,
 pred_class_idx=res[0][1][args.bbox_index] I was getting accurate heatmaps. This was further improved by increasing the number of masks used.

**Comment by RuoyuChen10** [comment]
> Hi, as suggested by you, I varied the patch sizes and number of patches but that was not the reason for poor heatmaps.
> 
> In drise-yolov3.py when the function call to generate_saliency_map is made the target class index was hardcoded to 1.
> 
> ```
> feat = model.extract_feat(data['img'][0].cuda())
> res = model.bbox_head.simple_test(
>     feat, data['img_metas'][0], rescale=True)
> target_box = res[0][0][args.bbox_index][:-1].cpu().detach().numpy().astype(np.int32)
> 
> 
> saliency_map = generate_saliency_map(model,
>                                     image,
>                                     target_class_index=1, #hardcoded
>                                     target_box=target_box,
>                                     prob_thresh=0.5,
>                                     grid_size=(16, 16),
>                                     n_masks=1000)
> ```
> 
> So, on passing the target class index with the index of the predicted class, pred_class_idx=res[0][1][args.bbox_index] I was getting accurate heatmaps. This was further improved by increasing the number of masks used.

Oh, yes, this is a bug, thanks.
